Sources/GRPC/Box.swift

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+
+/// Provides a "box" to put a value in.
+///
+/// Allows large values to be passed around without being copied.
+public final class Box<T> {
```

Comment:
do you really want this to be public?

Reply:
Not really, but it's used by a few public enums which are used in public handlers. I'll prefix it with an underscore.